### DISCUSSION TITLE
v-model not type checked in template

### DISCUSSION THREAD

**Discussion by Quartoffel** [vuejs/language-tools#4529] [Q&A] [answered]
Hey,

I have a problem that my `v-model` (or any v-* for that matter) are neither type checked nor show any intellisense information. Normal props work fine (even :model-value). I am not sure why this happens. Maybe someone has any ideas how to debug/fix this. Here are two pictures to show the issue in action:

1. Here it correctly throws an error because of type mismatch
![Correctly checks the value](https://github.com/vuejs/language-tools/assets/174067704/5a3bf48c-0b56-4e80-b690-ddedfdf6c22e)

2. Here it does not (and also does not show any Intellisense on hover):
![Does not check model-value](https://github.com/vuejs/language-tools/assets/174067704/7f4a2708-7ca7-4c4b-982e-821804f778f1)

Test2.vue is just:
```
<template>
  <div>{{ model }}</div>
</template>

<script setup lang="ts">
const model = defineModel<string>();
</script>
```

Information about my setup:
 System:
    OS: Windows 10 10.0.19045
    CPU: (12) x64 AMD Ryzen 5 5600X 6-Core Processor
    Memory: 16.40 GB / 31.93 GB
  Binaries:
    Node: 20.11.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 10.2.4 - C:\Program Files\nodejs\npm.CMD
    pnpm: 9.4.0 - ~\AppData\Local\pnpm\pnpm.CMD
  Browsers:
    Edge: Chromium (126.0.2592.87)
    Internet Explorer: 11.0.19041.4355

Vue - Official/vue-tsc Version: 2.0.26
Vue: 3.4.31
Typescript: 5.5.3

**Comment by davidmatter** [comment, marked as answer]
Created issue https://github.com/vuejs/language-tools/issues/4540 - please track there
